subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c

```diff
@@ -97,6 +100,29 @@ enum ec_host_cmd_status ec_host_cmd_send_in_progress_status(void)
 
 	return ret;
 }
+
+enum ec_host_cmd_status ec_host_cmd_send_in_progress_continue(ec_host_cmd_in_progress_cb_t cb,
+							      void *user_data)
+{
+	if (cmd_in_progress) {
+		return EC_HOST_CMD_BUSY;
+	}
+
+	cmd_in_progress = true;
```

Comment:
We should assign `saved_status = EC_HOST_CMD_UNAVAILABLE` here, not in `ec_host_cmd_send_in_progress_status()`.

- If the consumer doesn't check the result and runs another command that returns IN_PROGRESS, the old result can be read as the result of another command.
- In SMP systems, it's possible that `ec_host_cmd_send_in_progress_status()` (that runs on HOSTCMD thread) can override saved status, just after the result was saved in `handler_in_progress()` (that runs on sysworkq thread).

Reply:
Good point, done.